### ISSUE TITLE
Garbled standard error output while shell is initializing

### ISSUE THREAD

**Issue by josevalim** [erlang/otp#6263]
**Describe the bug**

See those shell snippets:

```
$ erl -noshell -eval 'shell:start_interactive({io, put_chars, [standard_error, "oops\noops"]}).'
Erlang/OTP 26 [DEVELOPMENT] [erts-13.0.4] [source-098208c93f] [64-bit] [smp:10:10] [ds:10:10:10] [async-threads:1] [jit]

oops
    oops*** ERROR: Shell process terminated! (^G to start new job) ***
```

```
$ erl -noshell -eval 'shell:start_interactive(), io:put_chars(standard_error, "oops\noops").'
Erlang/OTP 26 [DEVELOPMENT] [erts-13.0.4] [source-098208c93f] [64-bit] [smp:10:10] [ds:10:10:10] [async-threads:1] [jit]

oops
    oops*** ERROR: Shell process terminated! (^G to start new job) ***
```

**Affected versions**

While this error happens in previous Erlang/OTP versions, it is easier to reproduce as of the master branch with the new shell. Historically, Elixir has chosen to pass its `erl -user` process to work around this bug. However, if this is addressed, then we can rely exclusively on the new `shell:start_interactive/1` API.

Maybe this could be addressed by invoking the shell callback AND returning from `shell:start_interactive/1` only after the `standard_error` has been replaced?

**Comment by garazdawi** [comment]
I think #6266 fixes this problem.

However, as I was thinking about this I realized that you may want to be able to start in "oldshell" using `shell:start_interactive/1`. Do you know if IEx would ever need to do this?

**Comment by josevalim** [comment]
> However, as I was thinking about this I realized that you may want to be able to start in "oldshell" using shell:start_interactive/1

In my prototype, if `shell:start_interactive/1` returns enotsup, then I assume `user:start()` is the one running and I just start my shell directly. So from this perspective, my basis are covered. :)

**Comment by josevalim** [comment]
Closing this as it has been addressed in #6266 alongside other stuff. :)
